test/Removal.test.ts

```diff
@@ -0,0 +1,162 @@
+import type { BigNumber } from 'ethers';
+
+import type { Contracts } from '@/test/helpers';
+import { getDeployments, expect, hardhat } from '@/test/helpers';
+import { hre } from '@/utils/hre';
+
+const setupTest = hre.deployments.createFixture(
+  async (): Promise<Contracts> => {
+    await hre.deployments.fixture(); // ensure you start from a fresh deployments
+    return getDeployments({ hre });
+  }
+);
+
+describe('Removal', () => {
+  describe('Minting removals', () => {
+    it('should mint a batch of removals without listing any', async () => {
+      const { fifoMarket, removal } = await setupTest();
+      const { supplier } = await hardhat.getNamedAccounts();
+
+      const removalBalances = ['100', '200', '300', '400'];
+      const expectedMarketSupply = '0';
+      const removalVintages = [2018, 2019, 2020, 2021];
+      const listNow = false;
+      const packedData = hardhat.ethers.utils.defaultAbiCoder.encode(
+        ['address', 'bool'],
+        [fifoMarket.address, listNow]
+      );
+
+      const transactionResponse = await removal.mintBatch(
+        supplier,
+        removalBalances.map((balance) =>
+          hardhat.ethers.utils.parseUnits(balance)
+        ),
+        removalVintages,
+        packedData
+      );
+
+      const transactionReceipt = await transactionResponse.wait();
+      const batchMintedEvent = transactionReceipt?.events?.filter((event) => {
```

Comment:
Why wouldn't these events fire? Wouldn't we want this to throw if they didn't fire since if they didn't we' could end up with a false positive in this test run?